### ISSUE TITLE
Is this implementation correct?

### ISSUE THREAD

**Issue by Hhhhhhao** [vandit15/Class-balanced-loss-pytorch#4]
Hi, I wonder if this implementation is correct?

When I run the code with softmax, I get weights like this:
```
tensor([[0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [1.7646, 1.7646, 1.7646, 1.7646, 1.7646],
        [1.7646, 1.7646, 1.7646, 1.7646, 1.7646],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824],
        [0.8824, 0.8824, 0.8824, 0.8824, 0.8824]])
```

where each sample enjoin the same weights.

This is weird, right? According to the original paper, we should have different weight for each class (column), right?

**Comment by zdaiot** [comment]
@Hhhhhhhhhhao I have the same question. Can you help me?

**Comment by wmf1991yeah** [comment]
@Hhhhhhhhhhao I have the same question too. Why?

**Comment by Hhhhhhao** [comment]
sorry I didn't figure out yet...

**Comment by Saadmaghani** [comment]
I know this is an old issue but I just want to give my understanding to any one who has the same question (as I did too)

Yes, the columns should have different values. But since Cross Entropy and Focal Loss take the one-hot labels of the target and multiply it by this weight matrix, essentially the final result is that you are only taking the weight for a particular class. That particular class' weight is actually the one you see above, repeated along the entire column. 

Hope I helped!
